klippy/extras/display/display.py

```diff
@@ -232,16 +212,25 @@ def screen_update_128x64(self, eventtime):
         else:
             self.draw_status(0, 3, gcode_info, toolhead_info)
     # Screen update helpers
+    def draw_text(self, x, y, mixed_text, max_len=99):
+        pos = x
+        for i, text in enumerate(mixed_text.split('~')):
+            if pos >= max_len:
+                break
+            if i & 1 == 0:
+                # write text
+                self.lcd_chip.write_text(pos, y, text)
+                pos += len(text)
+            else:
+                # write glyph
```

Comment:
Hi Kevin, when testing the HD44780 I noticed that the bed temperature disappears when setting a target.  I believe it is happening here.  On line 101 self.draw_heater is called with an X position of 11.  From there, self.draw_text is called with max_len=8.  I believe this is causing the loop above to break before it draws the temperature.

UC1701 is behaving as expected though.

Reply:
Thanks.  My 2004 display is packed away right now so I didn't get a chance to test it.  The max_len check wasn't needed anyway, so I removed it from the code.  Hopefully should be fixed now.

-Kevin